ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.17__amd64-nightly-4.17-upgrade-from-stable-4.15.yaml

```diff
@@ -0,0 +1,197 @@
+  cron: 54 12 18 * *
+  steps:
+    cluster_profile: equinix-ocp-metal-qe
+    env:
+      DEVSCRIPTS_CONFIG: |
+        FIPS_MODE=true
+        IP_STACK=v4
+        NETWORK_TYPE=OVNKubernetes
+    test:
+    - as: set-upgrade-releases
+      commands: echo "${RELEASE_IMAGE_INTERMEDIATE416},${RELEASE_IMAGE_TARGET}" >
+        ${SHARED_DIR}/upgrade-edge
+      dependencies:
+      - env: RELEASE_IMAGE_INTERMEDIATE416
+        name: release:intermediate416
+      - env: RELEASE_IMAGE_TARGET
+        name: release:target
+      from: cli
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    - chain: openshift-upgrade-qe-test-chainupgrade
+    workflow: cucushift-installer-rehearse-baremetalds-ipi-ovn
+- as: ibmcloud-ipi-proxy-private-rt-f28
+  cron: 48 4 3 * *
+  steps:
+    cluster_profile: ibmcloud-qe
+    env:
+      RT_ENABLED: "true"
+    test:
+    - as: set-upgrade-releases
+      commands: echo "${RELEASE_IMAGE_INTERMEDIATE416},${RELEASE_IMAGE_TARGET}" >
+        ${SHARED_DIR}/upgrade-edge
+      dependencies:
+      - env: RELEASE_IMAGE_INTERMEDIATE416
+        name: release:intermediate416
+      - env: RELEASE_IMAGE_TARGET
+        name: release:target
+      from: cli
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    - chain: openshift-upgrade-qe-test-chainupgrade
+    workflow: cucushift-installer-rehearse-ibmcloud-ipi-proxy-private
+- as: nutanix-ipi-proxy-fips-mini-perm-f28
+  cluster: build01
+  cron: 57 11 7 * *
+  steps:
+    cluster_profile: nutanix-qe-dis
+    env:
+      EXTRACT_MANIFEST_INCLUDED: "true"
+      FIPS_ENABLED: "true"
+    test:
+    - as: set-upgrade-releases
+      commands: echo "${RELEASE_IMAGE_INTERMEDIATE416},${RELEASE_IMAGE_TARGET}" >
+        ${SHARED_DIR}/upgrade-edge
+      dependencies:
+      - env: RELEASE_IMAGE_INTERMEDIATE416
+        name: release:intermediate416
+      - env: RELEASE_IMAGE_TARGET
+        name: release:target
+      from: cli
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    - chain: openshift-upgrade-qe-test-chainupgrade
+    workflow: cucushift-installer-rehearse-nutanix-ipi-proxy-minimal-permission
+- as: vsphere-ipi-ovn-dualstack-privmaryv6-f28
+  cron: 36 18 26 * *
+  steps:
+    cluster_profile: vsphere-elastic
+    env:
+      IP_FAMILIES: DualStackIPv6Primary
+    test:
+    - as: set-upgrade-releases
+      commands: echo "${RELEASE_IMAGE_INTERMEDIATE416},${RELEASE_IMAGE_TARGET}" >
+        ${SHARED_DIR}/upgrade-edge
+      dependencies:
+      - env: RELEASE_IMAGE_INTERMEDIATE416
+        name: release:intermediate416
+      - env: RELEASE_IMAGE_TARGET
+        name: release:target
+      from: cli
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    - chain: openshift-upgrade-qe-test-chainupgrade
+    workflow: cucushift-installer-rehearse-vsphere-ipi-ovn-dualstack
+- as: vsphere-ipi-sdn-f28
```

Comment:
This one is duplicated with https://github.com/openshift/release/pull/53849/files#diff-a6c179159995b556d4a51c083fefb8b244160dcfe468467dd9a00f073ed844f6R92 ?

Reply:
oh...yes. but we have to:
1. cover sdn
2. cover all platform
and there is no better sdn profile in ocp 4.17 e2e job:
```
[jianl@jianl-thinkpadt14gen4 release]$ grep -e "- as: " ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__amd64-nightly.yaml | grep 'sdn'
- as: aws-ipi-private-fips-sdn-f28-disasterrecovery
- as: aws-ipi-proxy-sdn-workers-rhel8-fips-f14
- as: aws-ipi-proxy-sdn-workers-rhel8-fips-f28-destructive
- as: vsphere-ipi-sdn-migration-ovn-f14
[jianl@jianl-thinkpadt14gen4 release]$ 

[jianl@jianl-thinkpadt14gen4 release]$ grep -e "- as: " ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__arm64-stable.yaml | grep 'sdn'
[jianl@jianl-thinkpadt14gen4 release]$ 

[jianl@jianl-thinkpadt14gen4 release]$ grep -e "- as: " ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.15__multi-nightly.yaml | grep 'sdn'
- as: aws-ipi-sdn-migration-ovn-amd-f28
- as: azure-ipi-sdn-migration-ovn-amd-f28
- as: gcp-ipi-sdn-migration-ovn-arm-f28
[jianl@jianl-thinkpadt14gen4 release]$ 

```

Reply:
`vsphere-ipi-sdn` in 4.14->4.17 upgrade path already covered the config in 4.15->4.17, so I think they are duplicated.

Reply:
um, but we have to cover sdn in 4.15 -> 4.17 as well.

Reply:
Hmm, any specific reason?

Reply:
as discussed with @shellyyang1989 , upgrade from 4.14 -> 4.17 is different from 4.15 -> 4.17, so we'd better cover all key features in each version.

Reply:
My concern is that a cluster that is upgraded from a previous version might be not exact the same with a fresh installed cluster. So I don't think 4.y-1->4.y->4.y+1 is always able to cover 4.y->4.y+1. Just mho.

Reply:
If that is the case, the concern can be applied to all the configurations. We always can say the installation on some different source OCP version might be not exact the same though they are using the same configurations. How to handle other ci jobs?

And if I remember right, we made a decision, if a configuration is being covered in 4.14 -> 4.15 -> 4.16 -> 4.17 chain upgrade, then we should skip the same configuration in 4.15 -> 4.16 -> 4.17, because they are duplicated. Now changed the decision ?

Reply:
> the concern can be applied to all the configurations. 

This is why Jian Li said we need to cover all key features in all upgrade paths. if we have other sdn profile in 4.15, we can select a different one from the 4.14->4.17 for better coverage. otherwise, keeping it as is if fine to me as well.

Reply:
If we think vsphere-ipi-sdn is deserved to be covered in both 4.14->4.17 and 4.15->4.17, how to handle other configurations? Shall we also replicas them in all chain upgrade ci yaml ?

cc @jiajliu for insight on this topic.

Reply:
And back this specific profile, vsphere-ipi-sdn, what is the key feature in this coverage? SDN? A deprecating feature ?

Reply:
Yes, we intend to cover sdn here. It's a support exception in 4.15. As I said, we don't intend to cover one profile in both upgrade paths. If we have other sdn profile, we can switch.

Reply:
Okay, if you are confident with that it is a MUST, I will leave them to you, though I still can not figure out the test point why we need to cover the same `vsphere-ipi-sdn` in both 4.14->4.17 and 4.15->4.17.

Reply:
if only `vsphere-ipi-sdn` sdn profile is available for 4.15-4.17, how about changing the sdn profile on 4.14-4.17 path?

```
$ grep -e "- as: " ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.14__multi-nightly.yaml | grep 'sdn'
- as: aws-ipi-ovn-migration-sdn-amd-f28
- as: aws-ipi-ovn-migration-sdn-arm-f28
- as: aws-ipi-sdn-arm-f14
- as: aws-ipi-sdn-amd-f28-destructive
- as: aws-ipi-sdn-migration-ovn-amd-f14
- as: aws-ipi-sdn-migration-ovn-arm-f14
- as: aws-ipi-sdn-mtu-migrate-amd-f28
- as: aws-ipi-sdn-mtu-migrate-arm-f28
- as: azure-ipi-sdn-arm-f14
- as: azure-ipi-sdn-amd-f28-destructive
- as: azure-ipi-sdn-migration-ovn-amd-f14
- as: azure-ipi-sdn-migration-ovn-arm-f14
- as: gcp-ipi-sdn-amd-f28-destructive
- as: gcp-ipi-sdn-arm-f14
- as: gcp-ipi-sdn-migration-ovn-amd-f28
- as: gcp-ipi-sdn-migration-ovn-arm-f28
- as: gcp-upi-private-xpn-sdn-amd-f28-destructive
- as: gcp-upi-private-xpn-sdn-arm-f14
```

Reply:
I am fine with any other sdn profile, but then we have to find a new vsphere job to make sure all platform are covered.

Reply:
exchange `vsphere-ipi-sdn` in 4.14-4.17 amd64 job with `gcp-ipi-sdn-arm` in 4.14-4.17 multi job might resolve?

Reply:
since `gcp-ipi-sdn` could cover sdn, so +1 to use other vsphere profile to replace `vsphere-ipi-sdn`

Reply:
I want to replace `vsphere-ipi-sdn` with `vsphere-ipi-ovn-dualstack-privmaryv6`, and then remove `baremetalds-ipi-ovn-dualstack-primaryv6` from 4.14 -> 4.17. 
Do you think it is ok?

Reply:
fine with me, leave it you and others

Reply:
> I want to replace vsphere-ipi-sdn with vsphere-ipi-ovn-dualstack-privmaryv6, and then remove baremetalds-ipi-ovn-dualstack-primaryv6 from 4.14 -> 4.17.

@jianlinliu @shellyyang1989 any suggestion?

Reply:
it's fine with me. please also update the chain upgrade test sheet.

Reply:
done